internal/parser/scanner/scanner.go

```diff
@@ -228,23 +192,43 @@ func (s *scanner) peekString(str string) bool {
 }
 
 // createToken creates a token with given parameters
-func (s *scanner) createToken(t token.Type) token.Token {
-	tk := token.New(s.line, s.col, s.start, s.pos-s.start, t, string(s.input[s.start:s.pos]))
+func (s *scanner) createToken(t token.Type, value string) token.Token {
+	tk := token.New(s.line, s.col, s.start, s.pos-s.start, t, value)
 	s.start = s.pos
 	return tk
 }
 
-// seekNextPos returns the position of the end of a keyword.
+// seekTokenEnd returns the position of the end of a keyword.
 // It takes the start position of the keyword.
-func (s *scanner) seekNextPos(start int) int {
-	for start < len(s.input) && !whiteSpace.Matches(s.input[start]) {
+func (s *scanner) seekTokenEnd(start int) int {
+	for start < len(s.input) && !unicode.IsSpace(s.input[start]) { //!whiteSpace.Matches(s.input[start]) {
 		start++
 	}
 	return start
 }
 
+// unexpectedRune is returned when the combination/sequence of a set of
+// runes is not expected. The value argument contains all the valid
+// sequence of runes encountered till now and is consumed on execution
```

Comment:
the argument should probably be the expected (valid) runes, and the value should be an error message stating the error. similar to this.

`unexpected rune 'x', expected one of ['y', 'z']`

the parser can take this error message and refine it with the position information from the token. only then, this is a helpful error.

Reply:
The scanner currently dont know what runes it can be due to the usage of slices. 
I added the value string due to certain methods of logic, I probably can remove it from some refactoring and can add what _exactly_ the unexpected rune is.

Reply:
Yes, this's done.